### PR TITLE
New ACL for documents deleting

### PR DESCRIPTION
**Pull request by matrix-amiel**
Hi.

Right now only admin can delete documents.
Here I added a new ACL for document deleting, so another users could delete files with this permission. 
The default of this ACL is only admin.

Thanks
Amiel

### CODE REVIEW

File: acl_setup.php [hunk 1]
```diff
@@ -170,6 +170,8 @@
 // xl('Transactions (write optional)')
 $gacl->add_object('patients', 'Documents (write,addonly optional)', 'docs', 10, 0, 'ACO');
 // xl('Documents (write,addonly optional)')
+$gacl->add_object('patients', 'Documents deleting', 'rmdocs', 10, 0, 'ACO');
```
Comment (bradymiller) [inline acl_setup.php]: For consistency sake use:
Delete Documents
docs_rm

Comment (bradymiller) [reply acl_setup.php]: actually, do this instead:
Documents Delete
docs_rm

File: acl_upgrade.php [hunk 2]
```diff
@@ -681,14 +682,17 @@
     //Add new ACLs here (will return the ACL ID of newly created or already existant ACL)
     // (will also place in the appropriate group and CREATE a new group if needed)
     echo "<BR/><B>Adding ACLs(Access Control Lists) and groups</B><BR/>";
+    updateAcl($admin_write, 'Administrators', 'patients', 'Patients', 'rmdocs', 'Documents deleting', 'write');
+    updateAcl($emergency_write, 'Emergency Login', 'patients', 'Patients', 'rmdocs', 'Documents deleting', 'write');
```
Comment (bradymiller) [inline acl_upgrade.php]: For consistency sake:
Documents Delete
docs_rm

File: controllers/C_Document.class.php [hunk 3]
```diff
@@ -328,7 +328,7 @@ function view_action($patient_id = "", $doc_id)
 
         // Added by Rod to support document delete:
         $delete_string = '';
-        if (acl_check('admin', 'super')) {
+        if (acl_check('patients', 'rmdocs')) {
```
Comment (bradymiller) [inline controllers/C_Document.class.php]: For consistency sake:
docs_rm

**Comment by bradymiller** [comment]
Also, need to place a line in comments here with the new ACO:
https://github.com/openemr/openemr/blob/master/acl_upgrade.php#L98

**Comment by matrix-amiel** [comment]
Done.

**Comment by bradymiller** [comment]
hi @matrix-amiel ,
For consistency sake, change all the:
Documents deleting
to:
Documents Delete

**Comment by matrix-amiel** [comment]
Done.

**Comment by bradymiller** [comment]
Code looks good and will bring into codebase. Thanks for the improvement!
-brady

**Pull request merged**